tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -140,6 +219,74 @@ public string AsString(string defaultValue, Func<string, bool>? validator)
             return defaultValue.Result!;
         }
 
+        [return: NotNullIfNotNull(nameof(getDefaultValue))]
+        public T? GetAs<T>(Func<DefaultResult<T>>? getDefaultValue, Func<T, bool>? validator, Func<string, ParsingResult<T>> converter, string openTelemetryKey, Func<string, ParsingResult<T>> openTelemetryConverter)
+        {
+            var result = Source.GetAs<T>(Key, Telemetry, converter, validator, recordValue: true)
+                      ?? (FallbackKey1 is null ? null : Source.GetAs<T>(FallbackKey1, Telemetry, converter, validator, recordValue: true))
+                      ?? (FallbackKey2 is null ? null : Source.GetAs<T>(FallbackKey2, Telemetry, converter, validator, recordValue: true))
+                      ?? (FallbackKey3 is null ? null : Source.GetAs<T>(FallbackKey3, Telemetry, converter, validator, recordValue: true));
+
+            T? returnValue = default;
+            bool datadogConfigurationIsPresent = false;
+            if (result is { Result: { } value, IsValid: { } resultIsValid })
+            {
+                datadogConfigurationIsPresent = true;
+                if (resultIsValid)
+                {
+                    returnValue = value;
+                }
+            }
+            else
+            {
+                datadogConfigurationIsPresent = Source.IsPresent(Key)
+                                             || (FallbackKey1 is null ? false : Source.IsPresent(FallbackKey1))
+                                             || (FallbackKey2 is null ? false : Source.IsPresent(FallbackKey2))
+                                             || (FallbackKey3 is null ? false : Source.IsPresent(FallbackKey3));
+            }
+
+            // If there's a Datadog configuration present, check if a corresponding OpenTelemetry key is present so we can log the conflicting keys
+            if (datadogConfigurationIsPresent && Source.IsPresent(openTelemetryKey))
```

Comment:
Do we need to load otel config from everywhere, or is it explicitly just environment variables?